src/main/java/http/Router.java

```diff
@@ -0,0 +1,23 @@
+package http;
+
+import clientApplication.NotFound;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class Router implements Application {
+    public Map<RouteId, Application> routes;
+    public Router(Map<RouteId, Application> routes){
+        this.routes = routes;
+    }
+
+    public HttpResponse start(HttpRequest clientRequest){
+        RouteId id = new RouteId(clientRequest.getMethod(), clientRequest.getRequestPath());
+
+        Optional<Application> requestedApplication = Optional.ofNullable(routes.get(id));
+        Application verifiedApplication = requestedApplication.orElse(new NotFound());
+        HttpResponse applicationResponse = verifiedApplication.start(clientRequest);
+
+        return applicationResponse;
```

Comment:
*This is purely aesthetics*

I really like the way it reads when optionals are chained. Something like

```java
public HtttpResponse start(HttpRequest clientRequest) {
    RouteId id = new RouteId(clientRequest.getMethod(), clientRequest.getRequestPath());

    return Optional.ofNullable(routes.get(id))
        .orElse(new NotFound())
        .start(clientRequest);
}
```

It reads top to bottom in a way that is easy to understand. Bonus is that you don't have come up with a variable name for each intermediate stage (i.e. `verifiedApplication`).